### PR TITLE
Refactor engine.py to pull out some common functionality

### PR DESCRIPTION
**Pull request by mgoin**

### CODE REVIEW

File: src/deepsparse/engine.py [hunk 1]
```diff
@@ -152,7 +153,92 @@ def _validate_scheduler(scheduler: Union[None, str, Scheduler]) -> Scheduler:
...
+    def value(self):
+        return self._deepsparse_context
+
+    @property
+    def num_cores(self):
+        return self._num_cores
+
+    @property
+    def num_streams(self):
+        return self._deepsparse_context.num_streams()
```
Comment (bnellnm) [inline src/deepsparse/engine.py:193]: why use the deepsparse_context for one of these methods but not the other?

**Review by bnellnm — COMMENTED** [review]
Looks ok to me except for the stuff in KVCacheEngine

File: src/deepsparse/engine.py (hunk 1 above)
Comment (mgoin) [reply src/deepsparse/engine.py:193]: The context decides on the final number of streams so we need to ask it. We should store this value though

**Review by dbogunowicz — APPROVED** [review]
lgtm

**Pull request merged**